### PR TITLE
[quazip] conan v2 migration

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — APPROVED** [review]

File: recipes/quazip/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,10 @@
 cmake_minimum_required(VERSION 3.1)
```
Comment (toge) [inline recipes/quazip/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.8)
```
Please update `cmake_minimum_required`.
`target_compile_features` requires cmake/3.8.

File: recipes/quazip/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
-project(test_package)
+project(test_package CXX)
```
Comment (toge) [inline recipes/quazip/all/test_package/CMakeLists.txt:2]:
```suggestion
project(test_package LANGUAGES CXX)
```
It is trivial thing. You can miss it.

File: recipes/quazip/all/test_package/conanfile.py [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
 
     def build(self):
         cmake = CMake(self)
-        cmake.definitions["QT_VERSION_MAJOR"] = tools.Version(self.deps_cpp_info["qt"].version).major
         cmake.configure()
+        #cmake.definitions["QT_VERSION_MAJOR"] = tools.Version(self.deps_cpp_info["qt"].version).major
```
Comment (toge) [inline recipes/quazip/all/test_package/conanfile.py:21]:
```suggestion
```
Please remove comment out codes.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGtM

**Pull request merged**